pkg/util/machines.go

```diff
@@ -164,14 +162,19 @@ func GetMachineMetadata(hostname string, machine infrav1.VSphereVM, networkStatu
 			}
 		}
 		// check if DHCP is enabled
-		if machine.Spec.Network.Devices[i].DHCP4 {
+		if vsphereVM.Spec.Network.Devices[i].DHCP4 {
 			waitForIPv4 = true
 		}
-		if machine.Spec.Network.Devices[i].DHCP6 {
+		if vsphereVM.Spec.Network.Devices[i].DHCP6 {
 			waitForIPv6 = true
 		}
 	}
 
+	// Add the MAC Address to the network device
+	for i, status := range networkStatuses {
+		devices[i].MACAddr = status.MACAddr
```

Comment:
this also introduces the opposite issue: If someones adds a NIC manually, this would panic. `len(devices)` might be smaller than `len(networkStatuses)`